### ISSUE TITLE
Tests fail under pytorch 0.4

### ISSUE THREAD

**Issue by zou3519** [locuslab/qpth#12]
Hi,
The folks over at pytorch are working on cutting a new 0.4 release. We'd like to make the transition as smooth as possible, so we've been testing a number of community repos.

I ran into some test errors when running qpth tests against pytorch 0.4. Here is a repro script:
```
# Install pytorch-nightly (Currently our pre-release branch)
conda install pytorch-nightly -c pytorch

# Run tests
cd qpth/
nosetests -v -d test.py
```
The outputs of these tests are listed [in this gist](https://gist.github.com/zou3519/45f521db144e90627351a4a150b714f1). These failures could be problems with pytorch / I also might not be running them correctly, so I'm looking into them as well.

cc @soumith

**Comment by bamos** [comment]
Thanks for doing this for the 0.4 release! It's really helpful. I just fixed the issues and made sure they're backwards compatible and have pushed the latest version (the current master branch) to pypi. If you're curious, these issues are from 1) calling `.ndimension()` on an empty tensor used to return 0 but now it returns 1 (which I was able to correct in a backwards-compatible way by just checking `nelement()`) 2) dimension mis-match when adding or assigning tensors (this is great to have!)

The travis build is currently failing because of some dependencies and I'm not sure why, but the tests are indeed passing in older and newer versions of PyTorch.

-Brandon.
